src/lucky/routeable.cr

```diff
@@ -115,4 +145,81 @@ module Lucky::Routeable
       \{% raise "Use `route` instead of `with` if the action doesn't need params" %}
     end
   end
+
+  macro included
+    PARAM_DECLARATIONS = [] of Crystal::Macros::TypeDeclaration
+
+    macro inherited
+      PARAM_DECLARATIONS = [] of Crystal::Macros::TypeDeclaration
+    end
+  end
+
+  macro param(type_declaration)
+    {% PARAM_DECLARATIONS << type_declaration %}
+
+    def {{ type_declaration.var }} : {{ type_declaration.type }}
+      val = params.get(:{{ type_declaration.var.id }})
+      if val.nil?
+        nil || {{ type_declaration.value || nil }}
+      else
+        {% if type_declaration.type.class_name == "Union" %}
```

Comment:
This is the laymans way of doing a type check for `Union` types lol. There must be a better way? `union?` I can't seem to call even though I see it in the docs.